data-structures/linkedList/src/main/java/linkedList/LinkedList.java

```diff
@@ -220,4 +223,38 @@ public void insertAfter(int v, int newValue) {
         }
     }
 
+
+
+/****************
+ * KthFromTheEnd
+ *
+ * This method takes a value k and returns the node that is kth from the end of the linked list
+ * */
+    public int kthNodeFromTheEnd(int k) {
+        //try {
+        if (k > this.getSize() && this.getSize() > 1) throw new IndexOutOfBoundsException("The number provided is larger than the list.");
```

Comment:
Since the user doesn't know how long the LinkedList is, It's reasonable for the user to input a number greater than the LinkedList's length. Thusly, throwing an error when a user inputs a reasonable input seems a like a big punishment for a small infraction